### DISCUSSION TITLE
Diferent app hosts, diferent results

### DISCUSSION THREAD

**Discussion by Fabio-Vicente** [anuraghazra/github-readme-stats#2435] [Q&A] [answered]
Hello, guys!

Whats up?

I've deployed my own server for running Readme Stats, according instructions guidelines.

Howerver, something get a little bit weird to my app. Using the same query parameters that i've used on the main application, it generate LESS commits on my request:

**My application**
endpoint: https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&show_icons=true&include_all_commits=true&count_private=true&theme=ocean_dark
Seeing 378 commits

**Main application**
endpoint: https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&show_icons=true&include_all_commits=true&count_private=true&theme=ocean_dark
Seeing 451 commits

I understand that my application can show more information than the main application (and it's doing so in PRs and contributings) because it's use mt PAT, but never less, how is happening with my commits.

Obs: the settings for showing private contributions is on.

Additionaly, I have tested on another browser and even on another device to avoid cache outdated storage, but it given me the same result.

Any idea about what's going on?

Thanks!

**Comment by Zo-Bro-23** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/62234360/212934471-22b613cc-0fe2-43f8-83c8-8e317faf55d0.png)

Main endpoint.

![image](https://user-images.githubusercontent.com/62234360/212934513-a3e24bb3-4823-408a-9dd7-08faa22950cb.png)

Your endpoint.

Your endpoint shows higher contributions, which, like you said, is because it can count private contributions too. If it's showing differently for you, please include a screenshot. If not, it's likely something that happened when you first deployed your app, and it won't occur again. Cheers!

**Reply by rickstaa** [reply]
@Zo-Bro-23 is correct. Thank you for helping us maintain this repository! 🫂 The private instance has more permissions and would therefore show more contributions. However also remember we use CDN caching so it might take 4 hours for your card to update (see https://github.com/anuraghazra/github-readme-stats#common-options). You can break this cache by appending a random string `&random=hfjkbb`.

**Reply by Zo-Bro-23** [reply]
> You can break this cache by appending a random string `&random=hfjkbb`.

~You can also use the `cache_seconds` parameter right?~ (I was mistaken; the minimum is set as four hours, so using the param won't change anything)

> Thank you for helping us maintain this repository! :people_hugging:

You're welcome 😁 Any chance I can get Collaborator access though 😉 (for closing/labeling issues and reviewing/merging prs)? Perhaps after #2409 gets merged :)

**Reply by rickstaa** [reply]
Of course, I would appreciate all your help. I sent @anuraghazra a message you offered to help. I think #1867 is essential for maintaining this now very popular repository.

**Reply by Zo-Bro-23** [reply]
Thanks for the support :people_hugging:!

Also, time to close this discussion (already answered)?

**Reply by Fabio-Vicente** [reply]
Hello @Zo-Bro-23!

Thank you by answering.

Here is my screenshots:

![Captura de tela de 2023-01-20 01-47-01](https://user-images.githubusercontent.com/41867068/213619384-d14486dc-d609-4d0e-8c27-85652abe7c5d.png)
Main endpoint 

![Captura de tela de 2023-01-20 01-45-19](https://user-images.githubusercontent.com/41867068/213619149-ac30721e-c0b3-43f9-957a-cef2b09321fd.png)
My endpoint


My point is: the expected behavior should just be the opposite. My endpoint that could has more commits, not the main.

Notice that the card rendered on my computer in the main endpoint differs merely at the commits. Shouldn't It be the same by anyone who accessed else? I guessing it too weird...

Obs: I have accessed these endpoint just right now to take the screenshots.

Thanks by helping me discovering the mystery! 🙃

**Reply by Zo-Bro-23** [reply]
That's really weird. You're right that it should be the other way; your endpoint showing more, not less, commits than the main endpoint. Can you please verify token permissions by going to https://github.com/settings/tokens/? Since PRs and Contribs show more for your endpoint, my guess is that some repos are perhaps being discounted. Or maybe you committed using a different email, which doesn't show up on your endpoint (see [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)).

I still don't have a definitive answer for why this might be; I'm just as stumped as you are!

@rickstaa thoughts?

**Reply by Fabio-Vicente** [reply]
I have only this e-mail to contribute on GitHub and I've set my PAT to all repo permitions, as mentioned on deployment tutorial (actualy, I've set a few more read permitions to my PAT).

A new point at here is: I have some contributtions to organization privates repositories. Could it be the reason by all this mess?

**Reply by Zo-Bro-23** [reply]
> I have some contributtions to organization privates repositories. Could it be the reason by all this mess?

Might be! GitHub has two APIs for getting user info: one for the authenticated user (PAT) and the second one for other users. The main instance uses the second endpoint while your instance uses the first one (correct me if I'm wrong @rickstaa). This could very well be the reason for this discrepancy.

Again, I'm not exactly sure, and I'm just guessing. @rickstaa feel free to add to the discussion!

**Reply by rickstaa** [reply]
@Zo-Bro-23, @Fabio-Vicente, the problems you are experiencing could still be due to caching. GraphQL and especially the Rest API sometimes return slightly incorrect results which are then cached for 4 hours. 😅 The private instance shows fewer commits than the public one. Those instances use the same code, but in the private instance, you give it a PAT, which allows it to read your private repository data. Could you send me the URL of your private instance? 🤔 As explained in https://github.com/anuraghazra/github-readme-stats/discussions/1770#stats-card-related-questions you can also check the data returned from the GraphQL API yourself using the [graphql Explorer](https://docs.github.com/en/graphql/overview/explorer). The data received by our public Vercel instance is as follows:

```graphql
{
  "data": {
    "user": {
      "name": "Fábio Vicente",
      "login": "Fabio-Vicente",
      "contributionsCollection": {
        "totalCommitContributions": 110,
        "restrictedContributionsCount": 194
      }
    }
  }
}`
```

Which adds up to 304 commits.

**Reply by Zo-Bro-23** [reply]
The links are given [here](https://github.com/anuraghazra/github-readme-stats/discussions/2435#discussion-4765041) (original post).

Adding parameters to the URL will disable cache right? Or is that just the SVG cache and the stats are anyways cached for four hours?

Also, cache is likely to be the reason, since it was correct when I checked it a few days back (and was wrong a few days before that); see [here](https://github.com/anuraghazra/github-readme-stats/discussions/2435#discussioncomment-4707832).

**Reply by Zo-Bro-23** [reply]
https://github.com/anuraghazra/github-readme-stats/blob/8bc69e761ff7c922f47e6561d7a2f4fd0cc153c7/api/index.js#L64-L69

This seems to be the only code that caches stuff, and that sets the cache response header, which means that the browser will cache the SVG as is. This means that changing the request URL (adding/removing params) will cause everything to run again and get the correct values right? Am I missing something?

**Reply by rickstaa** [reply]
Correct the `&random=22341sdf` string, for example, will break the cache. This will allow us to compare the instances:

## Without include all commits and count private

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

## With include all commits and count private

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

## Without include all commits but with count private

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

## With both enabled

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&include_all_commits=true&count_private=true&random=232341sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

**Reply by rickstaa** [reply]
Looks like the public Vercel instance is down again. 😅 Let's wait till it shows up so that we can compare.

**Reply by rickstaa** [reply]
It looks like the Public instance is up again. The problem is caused by a bug in the `count_private` variable. I currently have not time to investigate this myself, but @Fabio-Vicente and @Zo-Bro-23 feel free to [create a bug report for it](https://github.com/anuraghazra/github-readme-stats/issues/new?assignees=&labels=bug&template=bug_report.yml).

![image](https://user-images.githubusercontent.com/17570430/213681204-c707519c-9766-411e-af42-2fd6ea5d0843.png)

**Comment by rickstaa** [comment]
@Fabio-Vicente I did some debugging, and it looks like it only happens on your fork; see the results below. Maybe you can try to redeploy your Vercel instance according to the `step-by-step guide at https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance. Feel free to comment below if things still happen after you re-deployed your instance. In that case, I think it is something in the GraphQL API that is going wrong with your account, and you should contact GitHub support.

![image](https://user-images.githubusercontent.com/17570430/214277065-e7c7f831-4901-4f51-aa8a-3a062e492922.png)

# Your user account

## Without count private

### My fork (I have no access to your private repositories)

[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Your fork (your PAT should have access to your private repositories)

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Public instance (should not have access to your private repositories)

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

## With count private

### My fork (I have no access to your private repositories)

[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Your fork (your PAT should have access to your private repositories)

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Public instance (should not have access to your private repositories)

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)


```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=Fabio-Vicente&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

# My user account

## Without count private

### My fork (I should have access to my private repositories)

[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Your fork (your PAT should not have access to my private repositories)

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Public instance (should not have access to my private repositories)

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=rickstaa&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

## With count private

### My fork (I should have access to my private repositories)

[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Your fork (your PAT should not have access to my private repositories)

[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://readme-stats-fabio-vicente.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```

### Public instance (should not have access to my private repositories)

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)


```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=rickstaa&count_private=true&random=2323431sd2f)](https://github.com/anuraghazra/github-readme-stats)
```
